File: Data Structures & Algorithms/balanced-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
    int dfs(TreeNode* root, bool &ans){
        if(!root or ans == false) return 0;
        int left_h = dfs(root->left, ans);
        int right_h = dfs(root->right,ans);
        if(abs(left_h - right_h)>1) ans = false;
        return 1 + max(right_h, left_h);
    }
public:
    bool isBalanced(TreeNode* root) {
        bool ans = true;
        dfs(root, ans);
        return ans;
    }
};

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.cs
public class Solution {
    public int[] ReplaceElements(int[] arr) {
        int max = arr[arr.Length-1];
        for(int i = (arr.Length - 1);i>=0;i--){
            int temp = arr[i];
            arr[i] = max;
            if(temp> max)
                max = temp;
        }
        arr[arr.Length-1] = -1;

        return arr;
    }
}